test-avocado/lib/host_setup.sh

```diff
@@ -0,0 +1,163 @@
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+
+
+# you have to setup domain in virt-manager default network and reboot the machine to apply changes
+# /etc/libvirt/qemu/networks/default.xml
+# add line
+# <domain name='cockpit.lan' localOnly='yes'/>
+unset command_not_found_handle
+HS_BASE_PCKGS="virt-deploy pystache sshpass telnet fabric python-pip avocado virt-manager qemu-img"
+export HS_GRP="virtualization"
+HS_CON="-c qemu:///system"
+export HS_POOLNAME="cockpitx"
+export HS_POOLNAME_PATH=/home/$HS_POOLNAME
+
+function echolog(){
+    echo "INFO: `date '+%Y%m%d-%H:%M:%S'`: $@"
+}
+
+function host_dependencies_fedora(){
+    sudo yum -y yum-plugin-copr
+    sudo yum -y copr enable fsimonce/virt-deploy
+    sudo yum -y copr enable lmr/Autotest
+    sudo yum -y install $HS_BASE_PCKGS
+}
+
+function host_dependencies_rhel7(){
+    sudo yum -y install https://dl.fedoraproject.org/pub/epel/7/x86_64/e/epel-release-7-5.noarch.rpm
+    curl https://copr.fedoraproject.org/coprs/fsimonce/virt-deploy/repo/epel-7/fsimonce-virt-deploy-epel-7.repo > virt-deploy.repo
+    curl https://copr.fedoraproject.org/coprs/lmr/Autotest/repo/fedora-21/lmr-Autotest-fedora-21.repo | sed -r -e 's/\$releasever/21/' -e 's/\$basearch/x86_64/' -e 's/gpgcheck=1/gpgcheck=0/' > avocado.repo
+    sudo /bin/cp virt-deploy.repo avocado.repo /etc/yum.repos.d/
+    sudo yum -y install $HS_BASE_PCKGS
+}
+function definepools(){
+    local HS_PNAME=$1
+    local HS_IPEXTENSION=123
+    sudo mkdir -p $HS_POOLNAME_PATH
+    sudo virsh $HS_CON pool-define /dev/stdin <<EOF
+<pool type='dir'>
+  <name>$HS_PNAME</name>
+  <target>
+    <path>$HS_POOLNAME_PATH</path>
+  </target>
+</pool>
+EOF
+    sudo virsh $HS_CON pool-start $HS_PNAME
+    sudo virsh $HS_CON pool-autostart $HS_PNAME
+
+    sudo virsh $HS_CON net-define /dev/stdin <<EOF
+<network>
+  <name>$HS_PNAME</name>
+  <domain name='cockpit.lan' localOnly='yes'/>
+  <forward mode='nat'/>
+  <bridge name='$HS_PNAME' stp='on' delay='0'/>
+  <mac address='52:54:00:AB:AB:AB'/>
+    <dns>
+   </dns>
+  <ip address='192.168.$HS_IPEXTENSION.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.$HS_IPEXTENSION.2' end='192.168.$HS_IPEXTENSION.254'/>
+    </dhcp>
+  </ip>
+</network>
+EOF
+    sudo virsh $HS_CON net-start $HS_PNAME
+    sudo virsh $HS_CON net-autostart $HS_PNAME
+    echolog "added network and storage pools"
+}
+
+function host_virtlibpolicy_solver(){
+    local LHS_USER=$HS_USER
+    definepools $HS_POOLNAME
```

Comment:
see https://github.com/jscotka/cockpit/blob/new_test_structure/test-avocado/lib/host_setup.sh it was repaired yesterday